lnbits/core/views/api.py

```diff
@@ -706,3 +709,29 @@ async def websocket_update_get(item_id: str, data: str):
         return {"sent": True, "data": data}
     except:
         return {"sent": False, "data": data}
+
+
+############################TINYURL##################################
+
+
+@core_app.post("/api/v1/tinyurl")
+async def api_create_tinyurl(url: str, endless: bool = False):
+    tinyurl = await get_tinyurl_by_url(url)
+    if tinyurl:
+        return tinyurl
+    return await create_tinyurl(url, endless)
```

Comment:
This has no authentication, is that intended?

Reply:
yes, I thought it would be more useful that way. Do you think its needed?

Reply:
Yep. Done for create, delete, and get tinyurls